apps/condo/domains/miniapp/schema/B2BAppPermission.js

```diff
@@ -58,7 +67,29 @@ const B2BAppPermission = new GQLListSchema('B2BAppPermission', {
                 },
             },
         },
+        name: {
+            schemaDoc: 'Name of permission. ' +
+                `Must be very brief (maximum ${MAX_PERMISSION_NAME_LENGTH} characters) ` +
+                'and describe the capabilities that the employee will get with this permission. ' +
+                'Must use nouns that indicate actions with objects or sections of the mini-application. ' +
+                'Examples: "Passes management", "Access to settings", "Contacts viewing".',
+            type: 'Text',
+            isRequired: true,
+            hooks: {
+                resolveInput: ({ resolvedData, fieldPath }) => {
+                    if (fieldPath in resolvedData) {
+                        return resolvedData[fieldPath].trim()
+                    }
 
+                    return undefined
+                },
+                validateInput: ({ resolvedData, fieldPath, context }) => {
+                    if (resolvedData[fieldPath].length > MAX_PERMISSION_NAME_LENGTH) {
```

Comment:
Is there a minimum length?

Reply:
Nope, there's not.

I've named this field `description` previously, but then I thought that it was too unclear, so I used field `shortDescription`. 

Then I thought, what is short? So I've added a limit.

And just before pushing when I was looking at designs again I thought that `name` option is the clearest one. But left old limitation :)

Reply:
That is, I can make an empty name? What will the user see in the interface?

Reply:
Refactored validation :)